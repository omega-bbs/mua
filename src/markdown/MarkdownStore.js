import { observable } from 'mobx'

class MarkdownStore {
  @observable files = null
  @observable mentions = null

  getFile(id) {
    if (!this.files) {
      return null
    }
    return this.files.find(item => item.id === id)
  }

  getMention(username) {
    if (!this.mentions) {
      return null
    }
    return this.mentions.find(item => item.username === username)
  }
}

export default MarkdownStore
