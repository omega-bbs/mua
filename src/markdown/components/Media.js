import React from 'react'
import PropTypes from 'prop-types'
import { inject, observer } from 'mobx-react'

import parseProtocol from '../../utils/parseProtocol'

@inject('markdownStore')
@observer
class Media extends React.Component {
  static propTypes = {
    markdownStore: PropTypes.object.isRequired,
    src: PropTypes.string.isRequired,
  }

  getFile() {
    const params = parseProtocol(this.props.src)
    if (!params || params.type !== 'file' || !params.id) {
      return null
    }
    return this.props.markdownStore.getFile(params.id)
  }

  render() {
    const file = this.getFile()
    if (!file) {
      return null
    }
    return null
  }
}

export default Media
