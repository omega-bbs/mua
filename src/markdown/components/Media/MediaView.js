import React from 'react'
import PropTypes from 'prop-types'

class MediaView extends React.Component {
  static propTypes = {
    file: PropTypes.object,
  }

  render() {
    return null
  }
}

export default MediaView
