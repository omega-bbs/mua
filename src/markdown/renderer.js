import ReactRenderer from 'commonmark-react-renderer'

const createRenderer = () => {
  return new ReactRenderer({
    skipHtml: true,
    softBreak: 'br',
  })
}

export default {
  renderer: null,

  render(ast) {
    if (!this.renderer) {
      this.renderer = createRenderer()
    }
    return this.renderer.render(ast)
  },
}
