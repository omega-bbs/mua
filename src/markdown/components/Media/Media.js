import React from "react";
import PropTypes from "prop-types";
import { inject, observer } from "mobx-react";

import MediaView from "./MediaView";
import parseProtocol from "../../../utils/parseProtocol";

@inject("markdownStore")
@observer
class Media extends React.Component {
  static propTypes = {
    markdownStore: PropTypes.object.isRequired,
    src: PropTypes.string.isRequired,
  };

  getFile() {
    const params = parseProtocol(this.props.src);
    if (!params || params.type !== "file" || !params.id) {
      return null;
    }
    return this.props.markdownStore.getFile(params.id);
  }

  render() {
    return <MediaView file={this.getFile()} />;
  }
}

export default Media;
