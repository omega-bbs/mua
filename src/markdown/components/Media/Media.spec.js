import React from "react";
import renderer from "react-test-renderer";

import MarkdownStore from "../../MarkdownStore";
import Media from "./Media";

jest.mock("./MediaView", () => {
  const React = require("react");
  const PropTypes = require("prop-types");

  class MediaView extends React.Component {
    static propTypes = {
      file: PropTypes.object,
    };

    render() {
      const file = this.props.file;
      if (!file) {
        return <mock-media-view>file not found</mock-media-view>;
      }
      return <mock-media-view>{`file #${file.id}`}</mock-media-view>;
    }
  }

  return MediaView;
});

const store = new MarkdownStore();
store.files = [{ id: "1" }];

const renderMedia = src => {
  return renderer.create(<Media markdownStore={store} src={src} />).toJSON();
};

test("basic", () => {
  expect(renderMedia("omega:file/1")).toMatchSnapshot();
});

test("not found", () => {
  expect(renderMedia("omega:file/2")).toMatchSnapshot();
});

test("error", () => {
  expect(renderMedia("omega:file")).toMatchSnapshot();
  expect(renderMedia("omega:unknown/1")).toMatchSnapshot();
  expect(renderMedia("http://example.com/image.png")).toMatchSnapshot();
});
