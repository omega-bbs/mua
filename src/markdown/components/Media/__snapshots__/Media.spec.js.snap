// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
<mock-media-view>
  file #1
</mock-media-view>
`;

exports[`error 1`] = `
<mock-media-view>
  file not found
</mock-media-view>
`;

exports[`error 2`] = `
<mock-media-view>
  file not found
</mock-media-view>
`;

exports[`error 3`] = `
<mock-media-view>
  file not found
</mock-media-view>
`;

exports[`not found 1`] = `
<mock-media-view>
  file not found
</mock-media-view>
`;
